app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1113,11 +1113,17 @@ private void sendMessage() {
                 return;
             } else {
                 try {
-                    Cursor streamCursor = makeStreamCursor(streamActv.getText().toString());
-                    if (streamCursor.getCount() == 0) {
+                    //check whether stream exists or not
+                    if (!isStreamExist(streamActv.getText().toString())) {
```

Comment:
Nice! the changes are looking good. 
I think we can improve performance of this code, instead of 2 database calls before sending message, we can use just isSubscribed() function.
It would return false in two case :
1) stream does not exist
2) stream not subscribed